website/source/docs/upgrading/plugins.html.md

```diff
@@ -0,0 +1,38 @@
+---
+layout: "docs"
+page_title: "Upgrading Plugins - Guides"
+sidebar_title: "Upgrade Plugins"
+sidebar_current: "docs-upgrading-plugins"
+description: |-
+  These are general upgrade instructions for Vault plugins.
+---
+
+# Upgrading Vault Plugins
+
+The following procedure details steps for upgrading a plugin that has already
+been registered to the catalog on a running server. This procedure is applicable
+to secret, auth, and database plugins.
+
+## Upgrade Procedure
+
+Vault executes plugin binaries when they are configured and roles are established
+around them. The binary cannot be modified or replaced while running, so
+upgrades cannot be performed by simply swapping the binary and updating the hash
+in the plugin catalog.
+
+Instead, you can restart or reload a plugin with the
+`sys/plugins/reload/backend` [API][plugin_reload_api]. Follow these steps to
+replace or upgrade a Vault plugin binary:
+
+1. Register plugin_v1 to the catalog
+2. Mount the plugin backend
+3. Register plugin_v2 to the catalog under the same plugin name, but with
+updated command to run plugin_v2 and updated sha256 of plugin_v2
+4. Trigger a plugin reload with sys/plugins/reload/backend to reload all mounted
```

Comment:
The instructions here differ slightly from https://www.vaultproject.io/api/system/plugins-reload-backend.html, which says exactly one of `plugin` or `mounts` must be provided.

Reply:
Should we recommend a plugin reload targeting only the upgraded plugin with e.g. [plugin](https://www.vaultproject.io/api/system/plugins-reload-backend.html#plugin) to keep things focused. What about changing step 4 to provide the `plugin` parameter instead of reloading all? It's possible that at the time I wrote the original KB, there was not yet a `plugin` parameter to that API.